plugins/keymanagerng/app/javascript/widgets/app/components/secrets/secretDetails.jsx

```diff
@@ -88,18 +89,44 @@ const SecretDetails = () => {
       secret?.data?.content_types?.default,
     ],
     queryFn: getSecretPayload,
-    enabled: !!payloadRequested,
+    enabled:
+      payloadRequested || // Enable when download requested
+      (![
+        "application/octet-stream",
+        "application/pkcs8",
+        "application/pkix-cert",
+      ].includes(secret?.data?.content_types?.default) &&
+        show), // Enable for specific content types when panel is shown
     onSuccess: (data) => {
-      const fileData = JSON.stringify(data)
-      const blob = new Blob([fileData], { type: "text/plain" })
-      const url = URL.createObjectURL(blob)
-      const link = document.createElement("a")
-      link.download = secret?.data?.name + ".json"
-      link.href = url
-      link.click()
+      if (
```

Comment:
Similar to above, it's easy to miss the not (`!`). Maybe change the if so that it doesn't check for what it is not, but what it is. So something like:
```javascript
if([...].includes(...)) {
  // Handle specific payload types by....
...
} else {
  // Handle payload types other than ...
...
}
```